File: 백준/Silver/2606. 바이러스/바이러스.py
from collections import deque

# 전체 노드 탐색 -> DFS 재귀함수 이용

#컴퓨터의 수 n
n = int(input())

#간선의 수 m
m = int(input())

graph = [[False] * (n+1) for i in range(0, n+1)]
for k in range(m):
    a, b = map(int, input().split())
    graph[a][b] = True
    graph[b][a] = True

#방문한 노드 기록
visited = [False] * (n + 1)
result = 0
def dfs(v):
    global result
    visited[v] = True
    result += 1
    for i in range(1, n + 1):
        if not visited[i] and graph[v][i]: #방문하지 않은 노드가 해당 노드와 연결이 되어 있다면
            dfs(i)

dfs(1)
print(result -1) #1번 컴퓨터 제외

File: 백준/Silver/2606. 바이러스/바이러스.java


import java.util.*;

public class Main {
    static ArrayList<Integer>[] graph;
    static boolean[] visited;
    static int count;

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();//컴퓨터 수
        int m= sc.nextInt();//연결된 컴퓨터 쌍의 수


        // 배열 할당
        graph = new ArrayList[n + 1];
        visited = new boolean[n + 1];

        //리스트 할당 (1번 노드부터 N번 노드)
        for(int i=1; i<n+1; i++) graph[i] = new ArrayList<>();

        for(int i = 0; i < m; i++){
            int a = sc.nextInt();
            int b = sc.nextInt();
            graph[a].add(b);// 연결된 노드
            graph[b].add(a);
        }

        dfs(1);
        System.out.println(count-1);



    }
    static void dfs(int node){
        visited[node] = true;
        count++;

        for(int next: graph[node]){
            if(!visited[next]) dfs(next); //방문하지 않은 노드 방문
        }
    }

}